From 832a9dabbfcb657bc67e649da017bfa7c065415f Mon Sep 17 00:00:00 2001
From: Vladimir Ermakov <vooon341@gmail.com>
Date: Wed, 10 Jul 2013 03:00:54 +0400
Subject: [PATCH 1/3] [ARM DT] i.MX6 Wandboard add sound stgl5000 and wifi+bt

stgl5000 connected to AUD3, I2C2.
Wifi thru SD2.
Bluetooth: UART3, AUD5.
Additional configuration for I2C1, I2C3.

Signed-off-by: Vladimir Ermakov <vooon341@gmail.com>
---
 arch/arm/boot/dts/imx6q-wandboard.dts | 183 +++++++++++++++++++++++++++++++++-
 arch/arm/boot/dts/imx6q.dtsi          |  58 +++++++++++
 2 files changed, 237 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/imx6q-wandboard.dts b/arch/arm/boot/dts/imx6q-wandboard.dts
index 24be415..b04217b 100644
--- a/arch/arm/boot/dts/imx6q-wandboard.dts
+++ b/arch/arm/boot/dts/imx6q-wandboard.dts
@@ -12,18 +12,153 @@
 	memory {
 		reg = <0x10000000 0x80000000>; /* 2GB */
 	};
+
+	regulators {
+		compatible = "simple-bus";
+
+		reg_2p5v: 2p5v {
+			compatible = "regulator-fixed";
+			regulator-name = "2P5V";
+			regulator-min-microvolt = <2500000>;
+			regulator-max-microvolt = <2500000>;
+			regulator-always-on;
+		};
+
+		reg_3p3v: 3p3v {
+			compatible = "regulator-fixed";
+			regulator-name = "3P3V";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-always-on;
+		};
+	};
+
+	sound {
+		compatible = "fsl,imx6q-wandboard-sgtl5000",
+			     "fsl,imx-audio-sgtl5000";
+		model = "imx6q-wandboard-sgtl5000";
+		ssi-controller = <&ssi1>;
+		audio-codec = <&codec>;
+		audio-routing =
+			"MIC_IN", "Mic Jack",
+			"Mic Jack", "Mic Bias",
+			"Headphone Jack", "HP_OUT";
+		mux-int-port = <1>;
+		mux-ext-port = <3>;
+	};
+
+	rfkill {
+		/* NOTE not fully functional rfkill */
+		compatible = "wand,imx6q-wandboard-rfkill";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_rfkill>;
+
+		bluetooth-on = <&gpio3 13 0>;
+		bluetooth-wake = <&gpio3 14 0>;
+		bluetooth-host-wake = <&gpio3 15 0>;
+
+		wifi-ref-on = <&gpio2 29 0>;
+		wifi-rst-n = <&gpio5 2 0>;
+		wifi-reg-on = <&gpio1 26 0>;
+		wifi-host-wake = <&gpio1 29 0>;
+		wifi-wake = <&gpio1 30 0>;
+	};
+};
+
+&iomuxc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_cko1_spdif &pinctrl_audmux_bt_aud5>;
+
+	uart1 {
+		pinctrl_uart1_rtscts: uart1grp-rtscts {
+			fsl,pins = <
+				MX6Q_PAD_EIM_D19__UART1_CTS_B 0x1b0b1
+				MX6Q_PAD_EIM_D20__UART1_RTS_B 0x1b0b1
+			>;
+		};
+	};
+
+	usdhc1 {
+		pinctrl_usdhc1_wand: usdhc1grp-wand {
+			fsl,pins = <
+				MX6Q_PAD_GPIO_2__GPIO1_IO02 0x1b0b0 /* Card detect GPIO1.2 */
+			>;
+		};
+	};
+
+	usdhc3 {
+		pinctrl_usdhc3_wand: usdhc3grp-wand {
+			fsl,pins = <
+				MX6Q_PAD_EIM_DA9__GPIO3_IO09 0x1b0b0 /* Card detect GPIO3.9 */
+			>;
+		};
+	};
+
+	rfkill {
+		pinctrl_rfkill: rfkillgrp-1 {
+			fsl,pins = <
+				/* Wi-Fi control pads */
+				MX6Q_PAD_EIM_EB1__GPIO2_IO29   0x1b0b1 /* ref_on, enable 32k clk */
+				MX6Q_PAD_EIM_A25__GPIO5_IO02   0x1b0b1 /* Wi-Fi reset (active low) */
+				MX6Q_PAD_ENET_RXD1__GPIO1_IO26 0x1b0b1 /* reg on for FDC6331L */
+				MX6Q_PAD_ENET_TXD1__GPIO1_IO29 0x1b0b1 /* host wake */
+				MX6Q_PAD_ENET_TXD0__GPIO1_IO30 0x1b0b1 /* wifi wake */
+				/* Bluetooth control pads */
+				MX6Q_PAD_EIM_DA13__GPIO3_IO13  0x1b0b1 /* bt_on */
+				MX6Q_PAD_EIM_DA14__GPIO3_IO14  0x1b0b1 /* bt_wake */
+				MX6Q_PAD_EIM_DA15__GPIO3_IO15  0x1b0b1 /* bt_host_wake */
+			>;
+		};
+	};
+
+	audmux {
+		pinctrl_cko1_spdif: cko1spdifgrp-1 {
+			fsl,pins = <
+				MX6Q_PAD_GPIO_0__CCM_CLKO1    0x10059
+				MX6Q_PAD_ENET_RXD0__SPDIF_OUT 0x10059
+			>;
+		};
+
+		pinctrl_audmux_bt_aud5: audmuxgrp-bt-aud5 {
+			fsl,pins = <
+				MX6Q_PAD_KEY_COL0__AUD5_TXC  0x80000000
+				MX6Q_PAD_KEY_ROW0__AUD5_TXD  0x80000000
+				MX6Q_PAD_KEY_COL1__AUD5_TXFS 0x80000000
+				MX6Q_PAD_KEY_ROW1__AUD5_RXD  0x80000000
+			>;
+		};
+	};
 };
 
 &fec {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_enet_1>;
+	pinctrl-0 = <&pinctrl_enet_3>;
 	phy-mode = "rgmii";
 	status = "okay";
 };
 
+&ssi1 {
+	fsl,mode = "i2s-slave";
+	status = "okay";
+};
+
+&audmux {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_audmux_2>;
+};
+
 &uart1 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart1_1>;
+	pinctrl-0 = <&pinctrl_uart1_1 &pinctrl_uart1_rtscts>;
+	fsl,uart-has-rtscts;
+	status = "okay";
+};
+
+&uart3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart3_1>;
+	fsl,uart-has-rtscts;
 	status = "okay";
 };
 
@@ -33,12 +168,52 @@
 
 &usdhc1 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_usdhc1_1>;
+	pinctrl-0 = <&pinctrl_usdhc1_1 &pinctrl_usdhc1_wand>;
+	cd-gpios = <&gpio1 2 0>;
+	vmmc-supply = <&reg_3p3v>;
+	status = "okay";
+};
+
+&usdhc2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc2_2>;
 	status = "okay";
 };
 
 &usdhc3 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_usdhc3_2>;
+	pinctrl-0 = <&pinctrl_usdhc3_2 &pinctrl_usdhc3_wand>;
+	cd-gpios = <&gpio3 9 0>;
+	vmmc-supply = <&reg_3p3v>;
 	status = "okay";
 };
+
+&i2c1 {
+	status = "okay";
+	clock-frequency = <400000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c1_1>;
+};
+
+&i2c2 {
+	status = "okay";
+	clock-frequency = <400000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c2_2>;
+
+	codec: sgtl5000@0a {
+		compatible = "fsl,sgtl5000";
+		reg = <0x0a>;
+		clocks = <&clks 169>;
+		VDDA-supply = <&reg_2p5v>;
+		VDDIO-supply = <&reg_3p3v>;
+	};
+};
+
+&i2c3 {
+	status = "okay";
+	clock-frequency = <400000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c3_2>;
+};
+
diff --git a/arch/arm/boot/dts/imx6q.dtsi b/arch/arm/boot/dts/imx6q.dtsi
index cfc44d8..85e6e03 100644
--- a/arch/arm/boot/dts/imx6q.dtsi
+++ b/arch/arm/boot/dts/imx6q.dtsi
@@ -161,6 +161,28 @@
 							MX6Q_PAD_RGMII_RX_CTL__RGMII_RX_CTL 0x1b0b0
 						>;
 					};
+
+					pinctrl_enet_3: enetgrp-3 {
+						fsl,pins = <
+							MX6Q_PAD_ENET_MDIO__ENET_MDIO       0x1b0b0
+							MX6Q_PAD_ENET_MDC__ENET_MDC         0x1b0b0
+							MX6Q_PAD_RGMII_TXC__RGMII_TXC       0x1b0b0
+							MX6Q_PAD_RGMII_TD0__RGMII_TD0       0x1b0b0
+							MX6Q_PAD_RGMII_TD1__RGMII_TD1       0x1b0b0
+							MX6Q_PAD_RGMII_TD2__RGMII_TD2       0x1b0b0
+							MX6Q_PAD_RGMII_TD3__RGMII_TD3       0x1b0b0
+							MX6Q_PAD_RGMII_TX_CTL__RGMII_TX_CTL 0x1b0b0
+							MX6Q_PAD_ENET_REF_CLK__ENET_TX_CLK  0x1b0b0
+							MX6Q_PAD_RGMII_RXC__RGMII_RXC       0x1b0b0
+							MX6Q_PAD_RGMII_RD0__RGMII_RD0       0x1b0b0
+							MX6Q_PAD_RGMII_RD1__RGMII_RD1       0x1b0b0
+							MX6Q_PAD_RGMII_RD2__RGMII_RD2       0x1b0b0
+							MX6Q_PAD_RGMII_RD3__RGMII_RD3       0x1b0b0
+							MX6Q_PAD_RGMII_RX_CTL__RGMII_RX_CTL 0x1b0b0
+						>;
+					};
+
+
 				};
 
 				gpmi-nand {
@@ -205,6 +227,13 @@
 							MX6Q_PAD_EIM_D16__I2C2_SDA 0x4001b8b1
 						>;
 					};
+
+					pinctrl_i2c2_2: i2c2grp-2 {
+						fsl,pins = <
+							MX6Q_PAD_KEY_COL3__I2C2_SCL 0x4001b8b1
+							MX6Q_PAD_KEY_ROW3__I2C2_SDA 0x4001b8b1
+						>;
+					};
 				};
 
 				i2c3 {
@@ -214,6 +243,13 @@
 							MX6Q_PAD_EIM_D18__I2C3_SDA 0x4001b8b1
 						>;
 					};
+
+					pinctrl_i2c3_2: i2c3grp-2 {
+						fsl,pins = <
+							MX6Q_PAD_GPIO_5__I2C3_SCL  0x4001b8b1
+							MX6Q_PAD_GPIO_16__I2C3_SDA 0x4001b8b1
+						>;
+					};
 				};
 
 				uart1 {
@@ -241,6 +277,17 @@
 					};
 				};
 
+				uart3 {
+					pinctrl_uart3_1: uart3grp-1 {
+						fsl,pins = <
+							MX6Q_PAD_EIM_D23__UART3_CTS_B   0x1b0b1
+							MX6Q_PAD_EIM_D24__UART3_TX_DATA 0x1b0b1
+							MX6Q_PAD_EIM_D25__UART3_RX_DATA 0x1b0b1
+							MX6Q_PAD_EIM_EB3__UART3_RTS_B   0x1b0b1
+						>;
+					};
+				};
+
 				uart4 {
 					pinctrl_uart4_1: uart4grp-1 {
 						fsl,pins = <
@@ -292,6 +339,17 @@
 							MX6Q_PAD_NANDF_D7__SD2_DATA7 0x17059
 						>;
 					};
+
+					pinctrl_usdhc2_2: usdhc2grp-2 {
+						fsl,pins = <
+							MX6Q_PAD_SD2_CMD__SD2_CMD    0x17059
+							MX6Q_PAD_SD2_CLK__SD2_CLK    0x10059
+							MX6Q_PAD_SD2_DAT0__SD2_DATA0 0x17059
+							MX6Q_PAD_SD2_DAT1__SD2_DATA1 0x17059
+							MX6Q_PAD_SD2_DAT2__SD2_DATA2 0x17059
+							MX6Q_PAD_SD2_DAT3__SD2_DATA3 0x17059
+						>;
+					};
 				};
 
 				usdhc3 {
-- 
1.8.1.2

